src/DocScan/Session/Retrieve/ResourceResponse.php

```diff
@@ -0,0 +1,50 @@
+<?php
+
+namespace Yoti\DocScan\Session\Retrieve;
+
+class ResourceResponse
+{
+
+    /**
+     * @var string|null
+     */
+    private $id;
+
+    /**
+     * @var TaskResponse[]
+     */
+    private $tasks = [];
+
+    /**
+     * ResourceResponse constructor.
+     * @param array<string, mixed> $resource
+     */
+    public function __construct(array $resource)
+    {
+        $this->id = $resource['id'] ?? null;
+
+        if (isset($resource['tasks'])) {
+            $parsed = [];
+            foreach ($resource['tasks'] as $task) {
+                $parsed[] = TaskResponseFactory::createTaskFromArray($task);
+            }
+            $this->tasks = $parsed;
+        }
+    }
+
+    /**
+     * @return string|null
+     */
+    public function getId(): ?string
+    {
+        return $this->id;
+    }
+
+    /**
+     * @return TaskResponse[]
+     */
+    public function getTasks(): array
+    {
+        return $this->tasks;
+    }
+}
```

Comment:
I think we also want `::getTextExtractionTasks()` here in case we introduce new tasks

Reply:
Fixed in 55a0dfa

Reply:
I can see this fixed in 77ce40a